### DISCUSSION TITLE
Allow multiple entities to be targeted by announcement service call

### DISCUSSION THREAD

**Discussion by pgale** [music-assistant/support#2538] [Feature requests and ideas]
Not sure if this is a but/feature request or the fault of something else, so wanted to ask here before raising it.

I have snapcast clients running on Android devices. I use the tts.speak or tts.cloud_say services to announce a variety of things. Sometimes while music playing, sometimes not. I have a group setup via HA group helper that contains all of the media players to target.

1) If I send a tts.speak to the group and a player is stopped/paused but has items in the queue, when the announcement has finished, playback of the queue starts. i.e. the announcement doesn't respect the previous state of the player when a group is targetted.

2) this is a different issue in that the announcements appear in the queue as an unknown artist and an 8 digit code. Is there a way to stop TTS announcements appearing in the queue.

Any thoughts? Thanks.

**Comment by OzGav** [comment]
What version of the integration are you using as the first one should have been fixed a few days ago?

**Comment by pgale** [comment]
2024.6.2

**Reply by pgale** [reply]
Individual targetted clients seems to work properly. This is just GROUPS of media players that are grouped in the HA helpers screen.

**Reply by OzGav** [reply]
What sort of groups?

**Reply by pgale** [reply]
As I mentioned above - "group setup via HA group helper that contains all of the media players to target". This means that I can target multiple media players in one go and add/remove players over time without having to go into each automation that calls for an announcement house-wide.

**Reply by OzGav** [reply]
Ok that is not supported. Grouping needs to be done in MA. You can use the HA service calls to join and unjoin like players together

**Reply by pgale** [reply]
So if I join players together, I can't control them separately and have to join and I join before/after each announcement? Not a good solution. 
Can you raise this as a feature request? It should be easy to target multiple players, particularly as the TTS commands don't have the option to target multiple layers in the UI.

**Reply by OzGav** [reply]
You can do multiple service calls as part of the automation.

**Reply by pgale** [reply]
Just not elegant.
